testsuite/tests/kuadrant/authorino/identity/test_extended_properties.py

```diff
@@ -0,0 +1,87 @@
+"""Basic tests for extended properties"""
+import base64
+import pytest
+from testsuite.objects import ExtendedProperty, Value
+from testsuite.httpx.auth import HeaderApiKeyAuth
+from testsuite.utils import extract_response
+
+EXTENDED_PROPERTIES = [
+    ExtendedProperty("property_static", Value(value="static")),
+    ExtendedProperty("property_dynamic", Value(jsonPath="context.request.http.path")),
+    ExtendedProperty("property_rewrite_true", Value(value="original")),
+    ExtendedProperty("property_rewrite_true", Value(value="rewritten"), overwrite=True),
+    ExtendedProperty("property_rewrite_false", Value(value="original")),
+    ExtendedProperty("property_rewrite_false", Value(value="rewritten"), overwrite=False),
+    ExtendedProperty("property_chain_static", Value(jsonPath="auth.identity.property_static")),
+    ExtendedProperty("property_chain_dynamic", Value(jsonPath="auth.identity.property_dynamic")),
+    ExtendedProperty("property_chain_self", Value(jsonPath="auth.identity.property_chain_self"), overwrite=True),
+]
+
+API_KEY_VALUE = "api_key_value"
+API_KEY_VALUE_B64 = base64.b64encode(API_KEY_VALUE.encode("ascii")).decode()
+
+
+@pytest.fixture(scope="module")
+def authorization(authorization):
+    """Add simple response to inspect 'auth.identity' part of authJson."""
+    authorization.responses.add_simple("auth.identity")
+    return authorization
+
+
+@pytest.fixture(scope="module")
+def auth_api_key(create_api_key, module_label):
+    """Creates API key Secret and returns auth for it."""
+    api_key = create_api_key("api-key", module_label, API_KEY_VALUE)
+    return HeaderApiKeyAuth(api_key)
+
+
+def assert_response(response):
+    """Check 'auth.identity.property_*' variables from extracted AuthJson."""
+    assert extract_response(response)["property_static"] % "" == "static"
+    assert extract_response(response)["property_dynamic"] % "" == response.url.path
+    assert extract_response(response)["property_rewrite_true"] % "" == "rewritten"
+    assert extract_response(response)["property_rewrite_false"] % "" == "original"
+    assert extract_response(response)["property_chain_static"] % 0 is None
+    assert extract_response(response)["property_chain_dynamic"] % 0 is None
+    assert extract_response(response)["property_chain_self"] % 0 is None
+
+
+def test_extended_properties(client, authorization, rhsso, module_label, auth, auth_api_key, blame):
+    """
+    This test checks extended properties function for sections in identity providers.
+    Currently test checks these identity sections:
+    - Anonymous
+    - API key
+    - Oidc
+
+    EXTENDED_PROPERTIES list contains testing properties added to every identity section.
+    """
+    authorization.identity.remove_all()  # clear before test
```

Comment:
The structure of this test seems weird:
1) ideally, you would not need to make changes to authorization after the setup
2) Seems like an ideal candidate for parametrization

Reply:
Parametrized as well

Reply:
No longer applicable